### PR TITLE
zcash_client_sqlite: Fix broken `account_birthday` method.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: zcash_client_sqlite/src/wallet.rs [hunk 1]
```diff
@@ -3201,4 +3203,15 @@ mod tests {
...
+
+    #[test]
+    fn test_account_birthday() {
+        let st = TestBuilder::new()
+            .with_block_cache()
+            .with_account_from_sapling_activation(BlockHash([0; 32]))
+            .build();
+
+        let account_id = st.test_account().unwrap().account_id();
+        assert_matches!(account_birthday(&st.wallet().conn, account_id), Ok(_))
```
Comment (str4d) [inline zcash_client_sqlite/src/wallet.rs:3215]: Check that it actually returns the correct birthday.

File: zcash_client_sqlite/src/wallet.rs [hunk 2]
```diff
@@ -1591,7 +1591,7 @@ pub(crate) fn account_birthday(
     conn.query_row(
         "SELECT birthday_height
          FROM accounts
-         WHERE account = :account_id",
+         WHERE id = :account_id",
```
Comment (str4d) [inline zcash_client_sqlite/src/wallet.rs:1594]: The account ID refactor, and the large-scale column name overhauling that was performed as a side-effect, has caused multiple of these bugs. Some of them we noticed and fixed in a hurry before 1.0, but like this one here, I do not doubt there are more.

What can we do to systematically check and fix all the queries, and ensure we don't have this problem in future? Defining a bunch of column name constants becomes unwieldy when it forces us to use string formatting, but if we start building wrappers to make that easier, we'd basically be building a custom DAO.

**Review by str4d — CHANGES_REQUESTED** [review]

**Comment by str4d** [comment]
Also, the fix needs to be documented in the changelog, as the bug made it into a published crate release.

File: zcash_client_sqlite/src/wallet.rs (hunk 2 above)
Comment (nuttycom) [reply zcash_client_sqlite/src/wallet.rs:1594]: Basically this just points to the fact that we actually need tests for all of the SQL queries; given that, such bugs become shallow.

**Comment by nuttycom** [comment]
[force-pushed](https://github.com/zcash/librustzcash/compare/8b390b988cd18632561e162c70fbfa09b48ddf74..f7c29f0) to address comments from code review.

**Review by str4d — APPROVED** [review]
utACK

File: zcash_client_sqlite/src/wallet.rs (hunk 2 above)
Comment (daira) [reply zcash_client_sqlite/src/wallet.rs:1594]: Is there some automated tool to lint / statically analyse SQL queries given a dump of the schema? It seems feasible.

**Review by daira — COMMENTED** [review]
Post-hoc ACK

**Pull request merged**